clar2wasm/tests/wasm-generation/main.rs

```diff
@@ -381,6 +382,63 @@ fn qualified_principal() -> impl Strategy<Value = Value> {
     })
 }
 
+fn value_to_string(value: &PropValue, to_response: bool) -> String {
+    match value.0 {
+        Value::Response(_) => format!("{}", value),
+        _ if to_response => format!("(ok {value})"),
+        _ => format!("{}", value),
+    }
+}
+
+fn ensure_last_response(last_value: PropValue, is_public_function: bool) -> PropValue {
+    if is_public_function {
+        match last_value.0 {
+            Value::Response(_) => PropValue::from(last_value.0),
+            _ => {
+                let val = Value::Response(ResponseData {
+                    committed: true,
+                    data: Box::new(last_value.0),
+                });
+                PropValue::from(val)
+            }
+        }
+    } else {
+        PropValue::from(last_value.0)
+    }
+}
+
+fn random_expressions(
```

Comment:
Here is the same, imagine being someone who didn't write this. What does `random_expression` even mean? Why are there 3 return values? `limit` is limit of what? Why a random expression should care if it is a public function?

Reply:
Okay there are two approaches. Increase the duplication a bit and or pass some vars to remove duplication. `limit` is maximum number of expressions being generated.

Reply:
Don't tell me what `limit` is: write a comment or make it more explicit!